lib/model/model.go

```diff
@@ -565,17 +569,18 @@ func (m *model) restartFolder(from, to config.FolderConfiguration, cacheIgnoredF
 }
 
 func (m *model) newFolder(cfg config.FolderConfiguration, cacheIgnoredFiles bool) error {
-	// Creating the fileset can take a long time (metadata calculation) so
-	// we do it outside of the lock.
+	m.mut.Lock()
+	defer m.mut.Unlock()
+
+	// Creating the fileset can take a long time (metadata calculation), but
+	// nevertheless should happen inside the lock.
```

Comment:
⚠️ I didn't fully investigate this, but literally ten lines above in a different method there's a comment that says we _must_ create filesets under the lock, so here we are.